Lib/fractions.py

```diff
@@ -247,12 +248,23 @@ def limit_denominator(self, max_denominator=1000000):
             n, d = d, n-a*d
 
         k = (max_denominator-q0)//q1
-        bound1 = Fraction(p0+k*p1, q0+k*q1)
-        bound2 = Fraction(p1, q1)
-        if abs(bound2 - self) <= abs(bound1-self):
-            return bound2
+        bound1_n = p0 + k*p1
+        bound1_d = q0 + k*q1
+        bound2_n = p1
+        bound2_d = q1
```

Comment:
We've lost some readability here: the original code clearly expressed the intent "if bound2 is closer to self than bound1 then return bound2, else return bound 1". With this code, I find it much harder to glean that intent.

A comment or two explaining what's going on might help.

I'd also suggest a bit of restructuring to keep things that are closely related together (and to keep consistency with the style of the rest of the function):
```suggestion
        bound1_n, bound1_d = p0 + k*p1, q0 + k*q1
        bound2_n, bound2_d = p1, q1
```
(And yes, there's some small overhead in the tuple packing and unpacking, but I think it's worth keeping for the readability.)